### PR TITLE
Add tests to check republish_values/propagate_host_value; fix bugs

### PR DESCRIPTION
**Pull request by justprosh**
- add initTopicAndSubscribeNode test
- add evict -> republish -> clear expired test
- fix bug with propagate_host_value and republish_values

### CODE REVIEW

File: service/src/impls.rs [hunk 1]
```diff
@@ -220,7 +222,7 @@ pub fn put_value_impl(key: String, value: String, current_timestamp_sec: u64, re
 
     check_key_existence(&connection, key.clone(), current_timestamp_sec.clone())?;
 
-    let values: Vec<Record> = get_values_helper(&connection, key.clone())?.into_iter().filter(|item| item.peer_id == item.set_by).collect();
+    let values: Vec<Record> = get_values_helper(&connection, key.clone())?.into_iter().filter(|item| item.peer_id != call_parameters.host_id).collect();
```
Comment (folex) [inline service/src/impls.rs:225]: Add a comment on what this filtering means, plz.

Comment (folex) [reply service/src/impls.rs:225]: Also, I'm not sure why you're loading all values. It seems that you're only using `values.len()` and `values.iter().last()`. Isn't it possible to do with SQL?

File: service/src/tests.rs [hunk 2]
```diff
@@ -771,4 +772,109 @@ mod tests {
...
+        assert!(result.success);
+        assert_eq!(result.result.len(), 1);
+        let record = result.result[0].clone();
+        assert_eq!(record.value, value);
+        assert_eq!(record.peer_id, HOST_ID);
+        assert_eq!(record.set_by, peer_id);
+    }
+
+    #[marine_test(config_path = "../Config.toml", modules_dir = "../artifacts/")]
+    fn evict_republish_clear_expired() {
```
Comment (folex) [inline service/src/tests.rs:823]: please add comments on steps in the test

File: service/src/tests.rs [hunk 3]
```diff
@@ -773,43 +773,53 @@ mod tests {
...
+        // make node a subscriber to it
+        let result = put_host_value_and_check!(aqua_dht, topic, value, timestamp, vec![], vec![], 0u32, cp);
         assert!(result.success);
 
         // clear db to imitate other node
         clear_env();
+
+        // notify neighbor about subscription
         let mut cp = get_correct_timestamp_cp(1);
         cp.init_peer_id = peer_id.clone();
```
Comment (folex) [inline service/src/tests.rs]: how does this code notify neighbour?

File: service/src/tests.rs [hunk 4]
```diff
@@ -773,43 +773,53 @@ mod tests {
...
         }];
-
         cp.init_peer_id = peer_id.clone();
 
+        // leave record about initial node subscription to topic
         let result = aqua_dht.propagate_host_value_cp(result, timestamp, weight.clone(), cp);
         assert!(result.success);
-        let result = aqua_dht.get_values_cp(key_str, 123u64, get_correct_timestamp_cp(1));
+
+        // check record existence
```
Comment (folex) [inline service/src/tests.rs:821]: does that mimicks `findSubscribers`?

File: service/src/tests.rs (hunk 3 above)
Comment (justprosh) [reply service/src/tests.rs]: https://github.com/fluencelabs/aqua-dht/blob/main/aqua/pubsub.aqua#L53
register topic and put value with `peer_id` = host_id

File: service/src/tests.rs (hunk 4 above)
Comment (justprosh) [reply service/src/tests.rs:821]: yeah, but for one node without merging

Comment (folex) [reply service/src/tests.rs:821]: mention that in the comment, plz

File: service/src/tests.rs [hunk 5]
```diff
@@ -846,32 +861,37 @@ mod tests {
...
 
+        // check values' existence
         let result = aqua_dht.get_values_cp(key.clone(), current_timestamp, get_correct_timestamp_cp(1));
         assert!(result.success);
         assert_eq!(result.error, "");
         assert_eq!(result.result.len(), 1);
 
-        let result = aqua_dht.clear_expired_cp(current_timestamp + DEFAULT_EXPIRED_VALUE_AGE, get_correct_timestamp_cp(0));
+        // increase timestamp to make value and key expired
+        let current_timestamp = current_timestamp + DEFAULT_EXPIRED_VALUE_AGE;
```
Comment (folex) [inline service/src/tests.rs:888]:
```suggestion
        let expired_timestamp = current_timestamp + DEFAULT_EXPIRED_VALUE_AGE;
```

**Review by folex — APPROVED** [review]

**Pull request merged**